pymongo/encryption.py

```diff
@@ -358,6 +358,16 @@ def __init__(self, kms_providers, key_vault_namespace, key_vault_client,
               - `aws`: Map with "accessKeyId" and "secretAccessKey" as strings.
                 These are the AWS access key ID and AWS secret access key used
                 to generate KMS messages.
+              - `azure`: Map with "tenantId", "clientId", and "clientSecret" as
+                strings. Additionally, "identityPlatformEndpoint" may also be
+                specified as a string (defaults to 'login.microsoftonline.com').
+                These are the Azure Active Directory credentials used to
+                generate Azure Key Vault messages.
+              - `gcp`: Map with "email" as a string and "privateKey" as a
+                base64 encoded string or `bytes`. Additionally,
+                "endpoint" may also be specified as a string (defaults to
+                'oauth2.googleapis.com'). These are the credentials used to
+                generate Google Cloud KMS messages.
               - `local`: Map with "key" as a 96-byte array or string. "key"
```

Comment:
We should mention base64 encoding here too (and in AutoEncryptionOpts):
```
"key" as `bytes` (96 bytes in length) or a base64 encoded string (unicode on Python 2) which decodes to 96 bytes.
```

Reply:
Done.